### ISSUE TITLE
It's not clear that the checkbox pattern requires a very specific markup

### ISSUE THREAD

**Issue by hatched** [canonical-web-and-design/vanilla-framework#3177]
## Pattern to amend

Checkbox

## Visual

![Screen Shot 2020-07-21 at 2 22 18 PM](https://user-images.githubusercontent.com/532033/88102974-a04cb480-cb5d-11ea-8e29-d6c0d80a707e.png)

## Context

In the above image the two column layout isn't possible without hacks for the  current implementation of the checkbox. As it requires that the `<label>` be an immediate sibling of the `<input>`.

**Comment by spencerbygraves** [comment]
The latest layout is at the top of [this PR](https://github.com/canonical-web-and-design/jaas-dashboard/pull/594). Best to talk to @hatched as I wasn't involved in the content.

**Comment by lyubomir-popov** [comment]
Ok, thnaks @spencerbygraves.

**Comment by lyubomir-popov** [comment]
@hatched interesting situation. This issue has two sides - 

1) The fact that we don't use any of the two [semantic](https://www.w3.org/WAI/tutorials/forms/labels/) ways to create a checkbox (label followed by checkbox, linked by the "for" attribute, or the "implicit" label wrapping the input). We don't because it is then impossible to style the checkbox, unless we have additional markup like a wrapper. We actually already have an validated WG  [issue ](https://github.com/canonical-web-and-design/vanilla-framework/issues/2931) to introduce wrappers, which may solve this partially.

2) Even if we had a wrapper, which would make it possible to visually rearrange the two regardless of the markup order, to implement the screenshot above we would need to match the column distribution of the parent layout.

@bartaz maybe markup like this where you can assign rol/col classes to parents of the label/input?
I've reference this issue in the checkbox pattern issue to make sure we cover this use case.

```
<div class="p-checkbox-wrapper row">
  <div class="col-2">
    <label for="checkExample1">HTML</label>
    <div class="col-4">
      <input type="checkbox" id="checkExample1">
    </div>
  </div>
</div>
```

**Comment by SirSamTumless** [comment]
Going to be fixed by #2931
